contracts/secret-storage/SecretManager.sol

```diff
@@ -515,39 +519,44 @@ contract SecretManager is
         uint256 storageTimeUsage;
         for (uint256 index = 0; index < len; index++) {
             uint256 secId = storeAckSecretIds[index];
-            uint256 enclaveIndex = _getSelectedEnclaveIndex(secId, enclaveAddress);
+            uint256 sizeLimit = userStorage[secId].sizeLimit;
+            storageTimeUsage += (_markEnclaveAlive(secId, enclaveAddress, _signTimestamp, lastAliveTimestamp, sizeLimit) * sizeLimit);
+        }
 
-            uint256 ackTimestamp;
-            if(_isReplacedStore(secId, enclaveIndex))
-                ackTimestamp = userStorage[secId].selectedEnclaves[enclaveIndex].replacedAckTimestamp;
-            else
-                ackTimestamp = userStorage[secId].ackTimestamp;
+        uint256 usdcPayment = storageTimeUsage * SECRET_STORE_FEE_RATE;
+        USDC_TOKEN.safeTransfer(owner, usdcPayment);
 
-            uint256 startTimestamp = ackTimestamp > lastAliveTimestamp ? ackTimestamp : lastAliveTimestamp;
-            uint256 endTimestamp = userStorage[secId].endTimestamp;
-            uint256 sizeLimit = userStorage[secId].sizeLimit;
+        emit SecretStoreAlive(enclaveAddress);
+    }
+
+    function _markEnclaveAlive(
+        uint256 _secretId,
+        address _enclaveAddress,
+        uint256 _signTimestamp,
+        uint256 _lastAliveTimestamp,
+        uint256 _sizeLimit
+    ) internal returns (uint256 /* storageTimeDuration */) {
+        uint256 enclaveIndex = _getSelectedEnclaveIndex(_secretId, _enclaveAddress);
 
-            // secret terminated
-            if(_signTimestamp > endTimestamp) {
-                SECRET_STORE.secretTerminationUpdate(enclaveAddress, sizeLimit, secId);
+        uint256 ackTimestamp = _getSecretStoreAckTimestamp(_secretId, enclaveIndex);
```

Comment:
ackTimestamp and endTimestamp is also getting accessed inside the _updateUsdcDepositPostPayment. Should optimise it? Will it reduce the gas?

Reply:
yes, updated it as it can save some gas.